src/main/java/org/cactoos/list/ImmutableListIterator.java

```diff
@@ -57,23 +57,23 @@ public ListIteratorOf(final List<T> list) {
      * @param list List that will be called to get a list iterator.
      * @param index Start index for a newly created list iterator.
      */
-    public ListIteratorOf(final List<T> list, final int index) {
+    public ImmutableListIterator(final List<T> list, final int index) {
         this(() -> list.listIterator(index));
     }
 
     /**
      * Ctor.
      * @param iter Original list iterator.
      */
-    public ListIteratorOf(final ListIterator<T> iter) {
+    public ImmutableListIterator(final ListIterator<T> iter) {
         this(() -> iter);
     }
 
     /**
      * Ctor.
      * @param orig Original list iterator.
      */
-    public ListIteratorOf(final Scalar<ListIterator<T>> orig) {
+    public ImmutableListIterator(final Scalar<ListIterator<T>> orig) {
         this.origin = new Unchecked<>(new Sticky<>(orig));
```

Comment:
@fanifieiev do you know why this takes a `Scalar`? I don't think `list.Immutable` does, why would `ImmutableListIterator` be different? It could make sense for `ListIteratorOf` to do so, but not for `ImmutableListIterator` I think

Reply:
@victornoel that's all fixed. Please have a look at newer commits